eng/testing/debug-dump-template.md

```diff
@@ -0,0 +1,122 @@
+# Debugging a CI dump
+
+This document describes how to debug a CI/PR test dump by downloading assets from helix, using a dotnet tool called `runfo`.
+
+## What is runfo?
+
+Runfo is a dotnet global tool that helps get information about helix test runs and azure devops builds. For more information see [this](https://github.com/jaredpar/runfo/tree/master/runfo#runfo)
+
+### How do I install it?
+
+You just need to run:
+
+```script
+dotnet tool install --global runfo
+```
+
+If you already have it installed, make sure you have at least version `0.6.1` installed, which contains support to download helix payloads. If you don't have the latest version just run:
+
+```script
+dotnet tool update --global runfo
+```
+
+## Download helix payload containing symbols:
+
+You can just achieve this by running:
+
+```script
+runfo get-helix-payload -j %JOBID% -w %WORKITEM% -o <out-dir>
+```
+
+> NOTE: if the helix job is an internal job, you need to pass down a [helix authentication token](https://helix.dot.net/Account/Tokens) using the `--helix-token` argument.
+
+This will download the workitem contents under `<out-dir>\workitems\` and the correlation payload under: `<out-dir>\correlation-payload\`. 
+
+> The correlation payload is usually the testhost or core root, which contain the runtime and dotnet host that we use to run tests.
+
+Once you have those assets, you will need to extract the testhost or core root. Then extract the workitem assets into the same location where coreclr binary is.
+
+## Windows dump on windows
+
+### Debug with WinDbg
+
+1. Install [dotnet-sos global tool](https://docs.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-sos).
```

Comment:
With the "internal" windbg sos.dll should be automatically be loaded from the extension gallery. windbg/cdb can be installed from \\dbg\privates\rc or \\dbg\privates\latest so installing dotnet-sos isn't necessary. It is a lot more manual process too, FYI, `dotnet tool install -g dotnet-sos`,  `dotnet-sos install` which displays the command too load this sos.dll like'.load C:\Users\mikem\.dotnet\sos\sos.dll'.

Linux dumps should also work with the latest windbg.

Reply:
I think we shouldn't add internal instructions here since this can also be done by contributors... but I'm fine if we add a note for that.

Reply:
Yes, I didn't think about extern contributors. They don't have access to this version of windbg and the extension gallery. So I guess you are on the right track with the more explicit instructions about installing dotnet-sos and explicitly loading sos. Just remember that in the internal windbg will already be already loaded with what should be the same version as dotnet-sos install.